Error message for `new Array(1.5)` is confusing

rkirsling opened the pull request:
#### 5fd85f75915069dee692d60e04cdf9479708e273
<pre>
Error message for `new Array(1.5)` is confusing
<a href="https://bugs.webkit.org/show_bug.cgi?id=262991">https://bugs.webkit.org/show_bug.cgi?id=262991</a>

Reviewed by Keith Miller.

Currently, `new Array(1.5)` gives the error &quot;Array size is not a small enough positive integer&quot;,
which is quite confusing since being &quot;small enough&quot; isn&apos;t the problem with 1.5 as an input.
Let&apos;s update this to &quot;Array length must be a positive integer of safe magnitude&quot;.

* JSTests/ChakraCore/test/es5/exceptions3.baseline-jsc:
* JSTests/stress/tail-call-host-call-throw.js:
* LayoutTests/js/array-from-expected.txt:
* LayoutTests/js/dom/dfg-negative-array-size-expected.txt:
* Source/JavaScriptCore/dfg/DFGOperations.cpp:
* Source/JavaScriptCore/runtime/ArrayConstructor.cpp:
* Source/JavaScriptCore/runtime/ArrayConstructor.h:
* Source/JavaScriptCore/runtime/CommonSlowPaths.cpp:

Canonical link: <a href="https://commits.webkit.org/269193@main">https://commits.webkit.org/269193@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/361e98bd64b2ea52cc19ace5c3c8fd4cf67c8ddf

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/21831 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/22055 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/22881 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/23704 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/20209 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/26287 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/22357 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/21321 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/22059 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/21646 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/18911 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/24556 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/18811 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/19769 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/26050 "Passed tests") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/18984 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/19840 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/19983 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/23915 "Passed tests") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/21200 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/20442 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/17/builds/17411 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/25246 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/19787 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/5961 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/5213 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/23995 "Built successfully") | | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/24/builds/26515 "Built successfully") | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/20382 "Built successfully") | | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/3/builds/5793 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

kmiller68 commented on Source/JavaScriptCore/runtime/ArrayConstructor.cpp:
```diff
@@ -79,7 +79,7 @@ JSArray* constructArrayWithSizeQuirk(JSGlobalObject* globalObject, ArrayAllocati
     uint32_t n = length.toUInt32(globalObject);
     if (n != length.toNumber(globalObject)) {
-        throwException(globalObject, scope, createRangeError(globalObject, "Array size is not a small enough positive integer."_s));
+        throwException(globalObject, scope, createRangeError(globalObject, "Array length must be a positive integer of safe magnitude."_s));
```
Can we share the error string in ArrayConstructor.h? Something like `constexpr ASCIILiteral arrayInvalidLengthError = "Array length must be a positive integer of safe magnitude."_s`

rkirsling replied:
Can do!

rkirsling committed:
Error message for `new Array(1.5)` is confusing
https://bugs.webkit.org/show_bug.cgi?id=262991

Reviewed by Keith Miller.

Currently, `new Array(1.5)` gives the error "Array size is not a small enough positive integer",
which is quite confusing since being "small enough" isn't the problem with 1.5 as an input.
Let's update this to "Array length must be a positive integer of safe magnitude".

* JSTests/ChakraCore/test/es5/exceptions3.baseline-jsc:
* JSTests/stress/tail-call-host-call-throw.js:
* LayoutTests/js/array-from-expected.txt:
* LayoutTests/js/dom/dfg-negative-array-size-expected.txt:
* Source/JavaScriptCore/dfg/DFGOperations.cpp:
* Source/JavaScriptCore/runtime/ArrayConstructor.cpp:
* Source/JavaScriptCore/runtime/ArrayConstructor.h:
* Source/JavaScriptCore/runtime/CommonSlowPaths.cpp:

Canonical link: https://commits.webkit.org/269193@main
